src/util/http_server.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use futures::sync::oneshot::{Receiver, Sender};
+use futures::{self, future, Future};
+use hyper::service::service_fn;
+use hyper::{self, Body, Error as HttpError, Method, Request, Response, Server, StatusCode};
+use prometheus::{self, Encoder, TextEncoder};
+use tokio_threadpool::{Builder, ThreadPool};
+
+use std::net::{AddrParseError, SocketAddr};
+use std::str::FromStr;
+
+quick_error! {
+    #[derive(Debug)]
+    pub enum Error {
+        AddrParse(err: AddrParseError) {
+            from()
+            cause(err)
+            display("{:?}", err)
+            description(err.description())
+        }
+        Http(err: HttpError) {
+            from()
+            cause(err)
+            display("{:?}", err)
+            description(err.description())
+        }
+    }
+}
+
+pub struct HttpServer {
```

Comment:
I think `tikv::server` mod is a better place for the http server.

Reply:
+1